File: Calculator.py
def add(x, y):
    return x + y

def subtract(x, y):
    return x - y

def multiply(x, y):
    return x * y

def divide(x, y):
    if y == 0:
        return "Error! Division by zero."
    return x / y

def calculator():
    while True:
        print("\nSelect operation:")
        print("A. Add")
        print("B. Subtract")
        print("C. Multiply")
        print("D. Divide")
        print("E. Exit")

        choice = input("Enter choice (A/B/C/D/E): ").upper()

        if choice == 'E':
            print("Exiting calculator. Goodbye!")
            break

        if choice not in ['A', 'B', 'C', 'D']:
            print("Invalid input. Please enter a letter between A and D.")
            continue

        try:
            num1 = float(input("Enter first number: "))
            num2 = float(input("Enter second number: "))
        except ValueError:
            print("Invalid number input. Please enter numeric values.")
            continue

        if choice == 'A':
            print(f"{num1} + {num2} = {add(num1, num2)}")
        elif choice == 'B':
            print(f"{num1} - {num2} = {subtract(num1, num2)}")
        elif choice == 'C':
            print(f"{num1} * {num2} = {multiply(num1, num2)}")
        elif choice == 'D':
            result = divide(num1, num2)
            print(f"{num1} / {num2} = {result}")

calculator()

calculator()

File: Calculator.dart
import 'dart:io';

void main() {
  while (true) {
    print('Enter first number:');
    double num1 = double.parse(stdin.readLineSync()!);

    print('Enter second number:');
    double num2 = double.parse(stdin.readLineSync()!);

    print('Choose an operation:');
    print('A: Add');
    print('B: Subtract');
    print('C: Multiply');
    print('D: Divide');

    String? choice = stdin.readLineSync();

    if (choice == 'A') {
      print('$num1 + $num2 = ${num1 + num2}');
    } else if (choice == 'B') {
      print('$num1 - $num2 = ${num1 - num2}');
    } else if (choice == 'C') {
      print('$num1 * $num2 = ${num1 * num2}');
    } else if (choice == 'D') {
      if (num2 != 0) {
        print('$num1 / $num2 = ${num1 / num2}');
      } else {
        print('Error: Cannot divide by zero.');
      }
    } else {
      print('Invalid choice.');
    }

    print('\nDo you want to perform another operation? (yes/no)');
    String? again = stdin.readLineSync();
    if (again?.toLowerCase() != 'yes') {
      break;
    }
  }
}              